### PR TITLE
[SHIPA-2940] Add CronJob to ketch jobs

### PR DESCRIPTION
**Pull request by stinkyfingers**
# Description

[SHIPA-2940] Adds Cronjob support. After discussing w/ Aleksej, I decided to extend the `Jobs CRD` rather than create a unique `CronJob CRD`. We were 50/50 on this decision, so I'm open to input preferring a `CronJob CRD`. 

Adds Cron-specific fields to `JobSpec`. Creates a K8s `CronJob` resource if the `schedule` string is populated. There is already a webhook assuring that 2 `jobs` do not have the same name. 

**Testing:**
Be sure to uninstall and `make generate manifest install` before running ketch.
`k apply -f cronjob.yaml`
`ketch job list`
`k get cronjobs -A`

sample yaml:
```
apiVersion: theketch.io/v1beta1
kind: Job
metadata:
  name: testcron
spec:
  name: testcron
  type: Job
  framework: test
  description: 'a test cron job - ketch'
  schedule: '* * * * *'
  containers:
    # list root dir
    - name: list
      image: ubuntu
      command:
        - ls
        - "/"
  policy:
    restartPolicy: Never
  completions: 1
  concurrencyPolicy: Forbid
  startingDeadlineSeconds: 10
  successfulJobsHistoryLimit: 2
  failedJobsHistoryLimit: 2
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (documentation addition or typo, file relocation)

## Testing

- [ ] New tests were added with this PR that prove my fix is effective or that my feature works (describe below this bullet)
- [ ] This change requires no testing (i.e. documentation update)

## Documentation

- [ ] All added public packages, funcs, and types have been documented with doc comments
- [ ] I have commented my code, particularly in hard-to-understand areas

## Final Checklist:

- [ ] I followed standard [GitHub flow](https://guides.github.com/introduction/flow/) guidelines
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

## Additional Information (omit if empty)

Please include anything else relevant to this PR that may be useful to know.

### CODE REVIEW

File: internal/templates/job/yamls/job.yaml [hunk 1]
```diff
@@ -27,4 +27,7 @@ spec:
           image: {{ $container.image }}
           command: {{ $container.command | toJson }}
         {{ end }}
+    metadata:
+      labels:
+        {{ $.Values.job.group }}/job-name: {{ $.Values.job.name | quote }}
```
Comment (stinkyfingers) [inline internal/templates/job/yamls/job.yaml:32]: Informers (e.g. in shipa) can use these labels to find a job's pods.

File: internal/templates/cronjob/yamls/cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          restartPolicy: {{ $.Values.job.policy.restartPolicy }}
+          containers:
+            {{ range $_, $container := $.Values.job.containers }}
+            - name: {{ $container.name }}
+              image: {{ $container.image }}
+              command: {{ $container.command | toJson }}
+            {{ end }}
+        metadata:
+          labels:
+            {{ $.Values.job.group }}/job-name: {{ $.Values.job.name | quote }}
```
Comment (stinkyfingers) [inline internal/templates/cronjob/yamls/cronjob.yaml:50]: Informers can use these labels to find a job's pods. CronJobs by default suffix `job-name` label values with a random number, so we need a custom label, like `theketch.io/job-name`.

File: internal/api/v1beta1/job_types.go [hunk 3]
```diff
@@ -68,6 +79,8 @@ type JobList struct {
 // Policy represents the policy types a job can have
 type Policy struct {
 	RestartPolicy RestartPolicy `json:"restartPolicy,omitempty"`
+	// CronJob-specific
```
Comment (aleksej-paschenko) [inline internal/api/v1beta1/job_types.go:82]: what do you think if we use RestartPolicy and ConccurentPolicy defined in go-client?

```golang

import (
	batchv1 "k8s.io/api/batch/v1"
	corev1 "k8s.io/api/core/v1"
)

// Policy represents the policy types a job can have
type Policy struct {
	RestartPolicy corev1.RestartPolicy `json:"restartPolicy,omitempty"`
	// CronJob-specific
	ConcurrencyPolicy batchv1.ConcurrencyPolicy `json:"concurrencyPolicy,omitempty"`
}
```

File: internal/chart/testdata/render_yamls/annotations-patch.yaml [hunk 4]
```diff
@@ -2,5 +2,5 @@
   path: /metadata/annotations/app
   value: "app1"
 - op: add
-  path: /metadata/annotations/shipa.io~1is-shipa
+  path: /metadata/annotations/theketch.io~1is-shipa
```
Comment (aleksej-paschenko) [inline internal/chart/testdata/render_yamls/annotations-patch.yaml:5]: I think, it is `theketch.io` in main. @kavinaravind merged a fix recently.

https://github.com/theketchio/ketch/blob/main/internal/chart/testdata/render_yamls/annotations-patch.yaml#L5

**Review by aleksej-paschenko — APPROVED** [review]
looks great!

**Pull request merged**